src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs

```diff
@@ -6,42 +6,67 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Remote;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
 {
     internal readonly struct UnitTestingRemoteHostClientWrapper
     {
-        internal UnitTestingRemoteHostClientWrapper(RemoteHostClient? underlyingObject)
+        internal UnitTestingRemoteHostClientWrapper(RemoteHostClient underlyingObject)
             => UnderlyingObject = underlyingObject;
 
         internal RemoteHostClient? UnderlyingObject { get; }
 
+        [MemberNotNullWhen(false, nameof(UnderlyingObject))]
         public bool IsDefault => UnderlyingObject == null;
 
         public static async Task<UnitTestingRemoteHostClientWrapper?> TryGetClientAsync(HostWorkspaceServices services, CancellationToken cancellationToken = default)
-            => new UnitTestingRemoteHostClientWrapper(await RemoteHostClient.TryGetClientAsync(services, cancellationToken).ConfigureAwait(false));
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(services, cancellationToken).ConfigureAwait(false);
+            if (client is null)
+                return null;
```

Comment:
📝 This is the only behavior change. Previously it returned a non-null object with `IsDefault` set, and now it returns `null`.

Reply:
@shyamnamboodiripad 
I remember in SBD/LUT it is checking the 'IsDefault' property,  we might need to change these checks to checking null after this PR is checking in?

Reply:
@Cosifne Looking at the references for [`TryGetClientAsync`](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=cn1hrhmmy4q1&file=ExternalAccess%5cUnitTesting%5cApi%5cUnitTestingRemoteHostClientWrapper.cs), all call sites are checking for both `HasValue` and `IsDefault`. So I think this should be ok.

Reply:
@shyamnamboodiripad that's exactly what I was hoping to hear!